src/main/python/ayab/engine/engine_fsm.py

```diff
@@ -46,7 +53,7 @@ class State(Enum):
     FINISHED = auto()
 
 
-class StateMachine(object):
+class StateMachine(QStateMachine):
```

Comment:
This is another mistake that the type checker caught- we were treating `StateMachine` like a `QStateMachine` but we weren't actually inheriting from it!